pandas/core/col.py

```diff
@@ -0,0 +1,293 @@
+
+    def __mul__(self, other: Any) -> Expr:
+        return self._with_binary_op("__mul__", other)
+
+    def __rmul__(self, other: Any) -> Expr:
+        return self._with_binary_op("__rmul__", other)
+
+    def __truediv__(self, other: Any) -> Expr:
+        return self._with_binary_op("__truediv__", other)
+
+    def __rtruediv__(self, other: Any) -> Expr:
+        return self._with_binary_op("__rtruediv__", other)
+
+    def __floordiv__(self, other: Any) -> Expr:
+        return self._with_binary_op("__floordiv__", other)
+
+    def __rfloordiv__(self, other: Any) -> Expr:
+        return self._with_binary_op("__rfloordiv__", other)
+
+    def __ge__(self, other: Any) -> Expr:
+        return self._with_binary_op("__ge__", other)
+
+    def __gt__(self, other: Any) -> Expr:
+        return self._with_binary_op("__gt__", other)
+
+    def __le__(self, other: Any) -> Expr:
+        return self._with_binary_op("__le__", other)
+
+    def __lt__(self, other: Any) -> Expr:
+        return self._with_binary_op("__lt__", other)
+
+    def __eq__(self, other: object) -> Expr:  # type: ignore[override]
+        return self._with_binary_op("__eq__", other)
+
+    def __ne__(self, other: object) -> Expr:  # type: ignore[override]
+        return self._with_binary_op("__ne__", other)
+
+    def __mod__(self, other: Any) -> Expr:
+        return self._with_binary_op("__mod__", other)
+
+    def __rmod__(self, other: Any) -> Expr:
+        return self._with_binary_op("__rmod__", other)
+
+    # Everything else
+    def __getattr__(self, attr: str, /) -> Callable[..., Expr]:
+        def func(df: DataFrame, *args: Any, **kwargs: Any) -> Any:
+            parsed_args = _parse_args(df, *args)
+            parsed_kwargs = _parse_kwargs(df, **kwargs)
+            return getattr(self(df), attr)(*parsed_args, **parsed_kwargs)
+
+        def wrapper(*args: Any, **kwargs: Any) -> Expr:
+            # Create a readable representation for method calls
+            args_repr = ", ".join(
+                repr(arg._repr_str if isinstance(arg, Expr) else arg) for arg in args
+            )
+            kwargs_repr = ", ".join(
+                f"{k}={v._repr_str if isinstance(v, Expr) else v!r}"
+                for k, v in kwargs.items()
+            )
+
+            all_args = []
+            if args_repr:
+                all_args.append(args_repr)
+            if kwargs_repr:
+                all_args.append(kwargs_repr)
+
+            args_str = ", ".join(all_args)
+            repr_str = f"{self._repr_str}.{attr}({args_str})"
+
+            return Expr(lambda df: func(df, *args, **kwargs), repr_str)
+
+        return wrapper
+
+    def __repr__(self) -> str:
+        return self._repr_str or "Expr(...)"
+
+    # Namespaces
+    @property
+    def dt(self) -> NamespaceExpr:
+        return NamespaceExpr(self, "dt")
+
+    @property
+    def str(self) -> NamespaceExpr:
+        return NamespaceExpr(self, "str")
```

Comment:
What if someone registers a custom series accessor, like `xyz` .  Is it possible to make `pd.col("a").xyz` work?  (not a big deal if not possible)

Reply:
Good call thanks, and supporting this leads to a simplification - have addressed, thanks!